src/interceptors/ClientRequest/utils/normalizeClientRequestArgs.test.ts

```diff
@@ -274,7 +274,7 @@ it('handles [RequestOptions, callback] input', () => {
   expect(url.href).toEqual('https://mswjs.io/resource')
 
   // Request options must be preserved.
-  expect(options).toEqual(initialOptions)
+  expect(options).toMatchObject(initialOptions)
```

Comment:
With the fix, the actual options object will contain _more_ things, like `_defaultAgent`. I suppose I good thing!